handlers/oam_handler.go

```diff
@@ -57,6 +58,15 @@ func (h *Handler) PatternFileHandler(
 		return
 	}
 
+	if r.Header.Get("Content-Type") == "application/json" {
+		body, err = yaml.JSONToYAML(body)
+		if err != nil {
+			rw.WriteHeader(http.StatusBadRequest)
+			fmt.Fprintf(rw, "failed to parse to PatternFile: %s", err)
```

Comment:
@utkarsh-pro are we ready for the new error code system?

(not correct, but an example of sorts)
```return errors.New(ErrHTTPStatusBadRequest, errors.Alert, []string{"failed to parse to PatternFile: %s", err.Error()}, []string{err.Error()}, []string{}, []string{})```